tools/cli/pkg/ers/command/migrate_all.go

```diff
@@ -115,18 +120,23 @@ func (c *MigrationAllCommand) simpleWorker(workerId int, workChannel chan ers.Wo
 			c.log.Infof("[Worker %d] %sInstance %s migrated%s",
 				workerId, Green, instance.Name, Reset)
 			c.wg.Done()
+			c.stats.Done()
 			continue
 		}
 		refreshed, err := c.ersClient.GetOne(instance.Id)
 		if err != nil {
 			c.log.Warnf("[Worker %d] GetOne error: %s", workerId, err.Error())
 			c.wg.Done()
+			c.stats.Err(instance.Id, err)
+			// TODO: add retries
```

Comment:
In case you need a snipped for a retry mechanism:

```
package main

import (
	"context"
	"errors"
	"fmt"
	"log"
	"time"
)


var count int

// Effector type defines a function signature.
type Effector func(context.Context) (string, error)

// Retry accepts an Effector and returns an anonymous function with the same signature as the Effector
// It essentially wraps the received function with the retry logic.
func Retry(effector Effector, retries int, delay time.Duration) Effector {
	return func(ctx context.Context) (string, error) {
		for r := 0; ; r++ {
			response, err := effector(ctx)
			if err == nil || r >= retries {
				return response, err
			}

			log.Printf("attempt %d failed; retrying in %v", r+1, delay)

			select {
			case <-time.After(delay):
			case <-ctx.Done():
				if ctx.Err() != nil {
					return "", fmt.Errorf("anonymous function from retry: %w", ctx.Err())
				}

				return "", nil
			}
		}
	}
}

func EmulateTransientError(ctx context.Context) (string, error) {
	count++

	if count <= 3 {
		return "intentional fail", errors.New("error")
	}

	return "success", nil
}

func main() {
	r := Retry(EmulateTransientError, 3, 1*time.Second)
	res, err := r(context.TODO())

	fmt.Println(res, err)
}



```